h/dynalloc.h

```diff
@@ -696,11 +751,24 @@ typedef struct DynallocInputParms_tag {
   char reserved[3];
 } DynallocInputParms;
 
+
+typedef struct DynallocNewTextUnit_tag {
```

Comment:
I don't know how I feel about this. It seems like we're adding another layer on top of DYNALLOC, which doesn't really simplify the use of DYNALLOC. How does this help use DYNALLOC? Would asking people to pass the original text units be basically the same?

Also, why is this "New" text units? Can they only be used to allocate data sets?